internal/pkg/configmanager/app.go

```diff
@@ -1,84 +1,118 @@
 package configmanager
 
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/devstream-io/devstream/pkg/util/log"
+	"github.com/devstream-io/devstream/pkg/util/mapz"
+	"github.com/devstream-io/devstream/pkg/util/scm"
+)
+
+const (
+	repoScaffoldingPluginName = "repo-scaffolding"
+)
+
 type (
-	App struct {
-		Name         string             `yaml:"name" mapstructure:"name"`
-		Spec         RawOptions         `yaml:"spec" mapstructure:"spec"`
-		Repo         *Repo              `yaml:"repo" mapstructure:"repo"`
-		RepoTemplate *RepoTemplate      `yaml:"repoTemplate" mapstructure:"repoTemplate"`
-		CIPipelines  []PipelineTemplate `yaml:"ci" mapstructure:"ci"`
-		CDPipelines  []PipelineTemplate `yaml:"cd" mapstructure:"cd"`
+	appRaw struct {
+		Name         string         `yaml:"name" mapstructure:"name"`
+		Spec         map[string]any `yaml:"spec" mapstructure:"spec"`
+		Repo         *scm.SCMInfo   `yaml:"repo" mapstructure:"repo"`
+		RepoTemplate *scm.SCMInfo   `yaml:"repoTemplate" mapstructure:"repoTemplate"`
+		CIRawConfigs []pipelineRaw  `yaml:"ci" mapstructure:"ci"`
+		CDRawConfigs []pipelineRaw  `yaml:"cd" mapstructure:"cd"`
 	}
-	Apps []App
+)
 
-	// AppInConfig is the raw structured data in config file.
-	// The main difference between App and AppInConfig is "CI" and "CD" field.
-	// The "CIRawConfigs" is the raw data of "CI" field defined in config file,
-	// which will be rendered to "CIPipelines" field in App with "PipelineTemplates".
-	// The "CDRawConfigs" is similar to "CIRawConfigs".
-	AppInConfig struct {
-		Name         string        `yaml:"name" mapstructure:"name"`
-		Spec         RawOptions    `yaml:"spec" mapstructure:"spec"`
-		Repo         *Repo         `yaml:"repo" mapstructure:"repo"`
-		RepoTemplate *RepoTemplate `yaml:"repoTemplate" mapstructure:"repoTemplate"`
-		CIRawConfigs []CICD        `yaml:"ci" mapstructure:"ci"`
-		CDRawConfigs []CICD        `yaml:"cd" mapstructure:"cd"`
+// getAppTools return app tools
```

Comment:
getAppTools is not very descriptive. I suggest a better name. For example, getToolsFromApp?